### ISSUE TITLE
Support for macros in krisajenkins/yesql

### ISSUE THREAD

**Issue by zoldar** [cursiveclojure/cursive#790]
Please consider adding support for yesql ( https://github.com/krisajenkins/yesql )? The query functions are generated by a macro. The query parameters are passed directly (positionally?) so the docstring is the only way to sort out what goes next in arguments list without looking into queries themselves.

**Comment by yatesco** [comment]
Just a minor point but the latest (beta) version of yesql supports named parameters as well.

**Comment by cursiveclojure** [comment]
Would it be acceptable to just show the query itself in the doc, rather than the yesql-generated docstring? That would mean that you could pop up the query at any time by using the quick documentation feature. Yesql appears to parse the queries itself, and I'd rather avoid adding instaparse as a dependency to Cursive.

**Comment by zoldar** [comment]
Colin Y.: Neat, didn't know about it. Thanks for pointing it out.

Anyway, it still would be useful to support it. Showing query itself is a
solution of sorts. It may be a bit more confusing than a straight docstring
when arguments get repeated and interleaved multiple times but still, it's
better than no docs at all.

Still, there will have to be some basic parsing done to extract query
strings (especially in the case of `defqueries` where multiple queries can
ba placed in a single file). But maybe it's simple enough that employing
the whole instaparse machinery won't be necessary.

2015-03-12 10:01 GMT+01:00 cursiveclojure <notifications@github.com>:

**Comment by cursiveclojure** [comment]
Yeah, I guess the petite queries they show in the examples are not typical in real projects. I'll think about this - it's relatively easy to support for IntelliJ Ultimate since that has full SQL support, but I'd like this to work in Community too.

**Comment by 42alex** [comment]
+1

**Comment by danielcompton** [comment]
Hugsql is a similar library, perhaps with more usage now?

**Comment by ikitommi** [comment]
+1 for HugSQL.

**Comment by kanwei** [comment]
+1 for hugsql

**Comment by zlosynus** [comment]
+1 for hugsql, it would even be helpful to just see which fns are defined in sql files.

**Comment by vitkalisz** [comment]
+1 for hugsql

**Comment by agrison** [comment]
+1 for HugSQL :) !

**Comment by ITSecMedia** [comment]
+1 for HugSQL :-D

**Comment by ClemRz** [comment]
+1 for HugSQL
